compiler/src/dotty/tools/dotc/typer/Nullables.scala

```diff
@@ -110,7 +110,11 @@ object Nullables with
     || { val sym = ref.symbol
          sym.is(Mutable)
          && sym.owner.isTerm
-         && sym.owner.enclosingMethod == curCtx.owner.enclosingMethod
+         && (if sym.owner != curCtx.owner then
```

Comment:
We can rewrite as
`sym.owner == curCtx.owner || (sym.owner.enclosingMethod == curCtx.owner.enclosingMethod && !curCtx.owner.is(Flags.Lazy))`

Could you add a comment explaining the two cases? Possibly with a short example code for each.

Reply:
I copied the section from the document